contrib-02-lhcb-openstack/Readme.md

```diff
@@ -0,0 +1,192 @@
+# LHCbSoftware on Openstack
+
+## Motivation
+
+The easiest way to to work in an LHCb Software Evironment is by logging in to an `lxplus` instance.
+From there one also has access to the [`AFS` file storage](http://information-technology.web.cern.ch/services/afs-service).
+
+However, working on `lxplus` comes with a few significant limitations:
+ * No super user access. This makes it hard to install new software.
+ * Shared resources with other users. Others might start resource-hungry processes on the same machine, therefore extending execution time and making it unpredicatable.
+
+Using a virtual machine on [`openstack.cern.ch`](https://openstack.cern.ch) solves these issues. It is possible to disposition a machine with up to 10 vCPUs and up to 20 GB memory. While vCPUs are not the same as actual, physical CPUs, this comes with with a significant improvement in performance and performance reliablity.
+However, the default VM images do not have the LHCb Software Environment installed, nor do they have access to AFS.
+
+This tutorial shows how one can easily deploy a machine on Openstack which offers the LHCb software environment and AFS access, using the CernVM virtual machine image.
+This can be done using the command line only. No step in the tutorial requires any GUI interaction.
+
+You can now deploy a powerful machine on openstack, run LHCb Software there while enjoying full super user access and access to AFS. You can take snapshots of your machines and if all goes awry, you can choose the nuclear option on your machine - and set up a new one.
+
+Especially useful for:
+  * Continous integration tests (CI)
+  * Profiling your algorithms.
+
+
+
+## Set up an Openstack instance with AFS and the LHCb Software environment
+
+
+This is a very basic tutorial, some steps described can be skipped if you have already done them.
```

Comment:
@apearce, @ibab, thoughts on this?
On Jun 17, 2016 09:37, "Konstantin Schubert" <notifications@github.com>
wrote:

> In contrib-02-lhcb-openstack/Readme.md
> <https://github.com/lhcb/impactkit-proceedings/pull/2#discussion_r67470405>
> :
>
> > +
> > +This tutorial shows how one can easily deploy a machine on Openstack which offers the LHCb software environment and AFS access, using the CernVM virtual machine image.
> > +This can be done using the command line only. No step in the tutorial requires any GUI interaction.
> > +
> > +You can now deploy a powerful machine on openstack, run LHCb Software there while enjoying full super user access and access to AFS. You can take snapshots of your machines and if all goes awry, you can choose the nuclear option on your machine - and set up a new one.
> > +
> > +Especially useful for:
> > +  * Continous integration tests (CI)
> > +  * Profiling your algorithms.
> > +
> > +
> > +
> > +## Set up an Openstack instance with AFS and the LHCb Software environment
> > +
> > +
> > +This is a very basic tutorial, some steps described can be skipped if you have already done them.
>
> I am not sure about the script. Id does't really automate a whole lot. In
> the end the whole procedure is just a few commands, especially after the
> image has been uploaded.
>
> I wonder if the script does more damage than good. Maybe people are better
> off integrating the deployment command and the keberos authentication for
> afs into their own scripts.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lhcb/impactkit-proceedings/pull/2/files/e8000eb534b1e0824961ba81fb04aa8fb79a3527#r67470405>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAqA8FSxq__JeXwtd9IwQJYjj834nEmvks5qMk4dgaJpZM4IlfJM>
> .
>